common/persistence/elasticsearch/esVisibilityStore.go

```diff
@@ -793,3 +811,46 @@ func checkPageSize(request *p.ListWorkflowExecutionsRequestV2) {
 		request.PageSize = 1000
 	}
 }
+
+func processAllValuesForKey(dsl *fastjson.Value, keyFilter func(k string) bool,
```

Comment:
Would be cool to have unit tests for all these awesome fined grained functions. 
The goal is to have as much coverage as possible, also easier for debugging when test fails.